playback/src/audio_backend/pipewire.rs

```diff
@@ -0,0 +1,338 @@
+                }
+            }
+            Ok(())
+        } else {
+            Err(PipeWireError::NotConnected.into())
+        }
+    }
+}
+
+impl Drop for PipeWireSink {
+    fn drop(&mut self) {
+        let _ = self.stop();
+    }
+}
+
+impl PipeWireSink {
+    pub const NAME: &'static str = "pipewire";
+}
+
+fn run_pipewire_loop(
+    consumer: RingConsumer,
+    quit_flag: Arc<AtomicBool>,
+    format: AudioFormat,
+) -> Result<(), PipeWireError> {
+    // Initialize PipeWire
+    pw::init();
+
+    let mainloop = pw::main_loop::MainLoopRc::new(None)
+        .map_err(|e| PipeWireError::MainLoopCreation(format!("{:?}", e)))?;
+
+    let context = pw::context::ContextRc::new(&mainloop, None)
+        .map_err(|e| PipeWireError::MainLoopCreation(format!("{:?}", e)))?;
+
+    let core = context
+        .connect_rc(None)
+        .map_err(|e| PipeWireError::MainLoopCreation(format!("{:?}", e)))?;
+
+    let stream = pw::stream::StreamBox::new(
+        &core,
+        "librespot-playback",
+        properties! {
+            *pw::keys::MEDIA_TYPE => "Audio",
+            *pw::keys::MEDIA_ROLE => "Music",
+            *pw::keys::MEDIA_CATEGORY => "Playback",
+            *pw::keys::AUDIO_CHANNELS => NUM_CHANNELS.to_string().as_str(),
+            *pw::keys::APP_NAME => "librespot",
+        },
+    )
+    .map_err(|e| PipeWireError::StreamCreation(format!("{:?}", e)))?;
+
+    let sample_size = calculate_sample_size(format);
+    let stride = sample_size * NUM_CHANNELS as usize;
+
+    // Clone mainloop for use in the listener callback
+    let mainloop_quit = mainloop.clone();
+
+    // Use PipeWire's real-time callback with lock-free ring buffer consumer
+    // This ensures optimal performance and real-time safety
+    let _listener = stream
+        .add_local_listener_with_user_data((consumer, quit_flag.clone()))
+        .process(move |stream, (consumer, quit_flag)| {
+            // Check if we should quit
+            if quit_flag.load(Ordering::Relaxed) {
+                mainloop_quit.quit();
+                return;
+            }
+
+            match stream.dequeue_buffer() {
+                None => {
+                    // No buffer available, this is normal
+                }
+                Some(mut buffer) => {
+                    let datas = buffer.datas_mut();
+                    let data = &mut datas[0];
+
+                    let n_frames = if let Some(slice) = data.data() {
+                        let n_frames = slice.len() / stride;
+                        let total_bytes = n_frames * stride;
+
+                        // Pop data from the lock-free ring buffer
+                        // This is wait-free and real-time safe
+                        let bytes_read = consumer.pop_slice(slice);
+
+                        // Fill any remaining space with silence if underrun
+                        if bytes_read < total_bytes {
+                            slice[bytes_read..total_bytes].fill(0);
```

Comment:
For `F32` and `F64` formats, I'm not sure if `u8` byte values of `0` are equal to `0.0` in float.